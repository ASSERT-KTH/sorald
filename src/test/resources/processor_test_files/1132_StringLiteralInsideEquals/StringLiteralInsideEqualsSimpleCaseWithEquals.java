/*
Test the simple case with equals(java.lang.Object): the target and the argument of the call should be swapped.
 */

public class StringLiteralInsideEqualsSimpleCaseWithEquals {
    public static void main(String[] args) {
        String myString = null;

        System.out.println("Equal? " + myString.equals("foo")); // Noncompliant; will raise a NPE
    }
}
