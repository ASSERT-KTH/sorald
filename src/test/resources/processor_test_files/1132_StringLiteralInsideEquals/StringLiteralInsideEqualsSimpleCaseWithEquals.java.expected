/*
Test the simple case with equals(java.lang.Object): the target and the argument of the call should be swapped.
 */

public class StringLiteralInsideEqualsSimpleCaseWithEquals {
    public static void main(String[] args) {
        String myString = null;

        System.out.println("Equal? " + "foo".equals(myString));
    }
}
